.github/workflows/mac_release.yml

```diff
@@ -8,22 +8,30 @@ on:
 env:
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   BUILD_TYPE: Release
-  FC: /usr/local/bin/gfortran-9
+  FC: gfortran-11
   SDKROOT: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
   Python_REQUIRED_VERSION: 3.8
 
 jobs:
   mac_release:
     runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.allow_failure }}
     strategy:
+      # fail-fast: Default is true, switch to false to allow one platform to fail and still run others
+      fail-fast: false
       matrix:
-        os: [macos-10.15, macos-11]
-        macos_dev_target: [10.15, 11.6]
-        exclude:
-          - os: macos-10.15
-            macos_dev_target: 11.6
-          - os: macos-11
-            macos_dev_target: 10.15
+        # TODO: macos-10.15 will be EOL on 8/30/22 and removed on 12/1/2022
+        os: [macos-10.15, macos-11, macos-12]
+        include:
+        - os: macos-10.15
+          allow_failure: true
+          macos_dev_target: 10.15
+        - os: macos-11
+          allow_failure: false
+          macos_dev_target: 11.6
+        - os: macos-12
+          allow_failure: false
+          macos_dev_target: 12.1
```

Comment:
allow 10.15 to fail without failing the job (for the day when it'll be actually removed).

For macos-12, the runner is on macOS 12.5, but I'm targetting 12.1 anyways.

I also switched to an `include`layout which is more versatile and clear than have a 3 x 3 = 9 matrix and making 6 exclusions...